File: src/ui/src/models/api/Action/UpsertActionModel.ts
import { BackendActionModelConfiguration } from './BackendActionModelConfiguration';
import { ToolProviderApplicationType } from '../provider/ToolProviderApplicationType';

/**
 * Model for creating or updating an action
 */
export interface UpsertActionModel {
  /**
   * The name of the action
   */
  name: string;
  
  /**
   * The description of the action
   */
  description: string;
  
  /**
   * The icon for the action
   */
  icon?: string;
  
  /**
   * The system prompt IDs associated with this action
   */
  systemPromptIds: string[];
  
  /**
   * The user prompt IDs associated with this action
   */
  userPromptIds: string[];
  
  /**
   * The tool provider application types that are allowed for this action
   */
  allowedTools: ToolProviderApplicationType[];
  
  /**
   * The action model configuration
   */
  actionModelConfiguration: BackendActionModelConfiguration;
}

File: src/backend/DonkeyWork.Chat.Api/Models/Action/UpsertActionModel.cs
// ------------------------------------------------------
// <copyright file="UpsertActionModel.cs" company="DonkeyWork.Dev">
// Provided as is by (c) DonkeyWork.Dev. No warranty or support is given.
// </copyright>
// ------------------------------------------------------

using DonkeyWork.Chat.Common.Models.Actions;
using DonkeyWork.Chat.Common.Models.Providers.Tools;

namespace DonkeyWork.Chat.Api.Models.Action;

/// <summary>
/// Model for creating or updating an action.
/// </summary>
public class UpsertActionModel
{
    /// <summary>
    /// Gets or sets the name.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the description.
    /// </summary>
    public string Description { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the icon.
    /// </summary>
    public string? Icon { get; set; }

    /// <summary>
    /// Gets or sets the system prompt ids.
    /// </summary>
    public List<Guid> SystemPromptIds { get; set; } = [];

    /// <summary>
    /// Gets or sets the user prompt ids.
    /// </summary>
    public List<Guid> UserPromptIds { get; set; } = [];

    /// <summary>
    /// Gets or sets the allowed tools.
    /// </summary>
    public List<ToolProviderApplicationType> AllowedTools { get; set; } = [];

    /// <summary>
    /// Gets or sets the action model configuration.
    /// </summary>
    public ModelConfiguration ModelConfiguration { get; set; } = new ();
}
